app/setup/index.js

```diff
@@ -0,0 +1,259 @@
+            // maybe in case they closed it or want to try again?
+            this.setState({
+                errorMessage: `We\'re waiting for authorization of [${username}].`});
+            this.verifyAuthDone();
+            if (this.state.loggedIn) {
+                // TODO: Is there a better way? On premise has the `external-data-connector`
+                // route so cant use window.location.origin.
+                clearInterval(checkAuth);
+                const appRoute = baseUrl().replace('/setup', '/');
+                window.location.assign(appRoute);
+            }
+        }, 1000);
+    }
+
+    authenticateUser () {
+        if (!this.state.username) {
+            this.setState({
+                status: 'failure',
+                errorMessage: 'Plotly username can not be empty.'
+            });
+            return;
+        }
+        if (!this.state.domain && this.state.serverType === ONPREM) {
+            this.setState({
+                status: 'failure',
+                errorMessage: 'Plotly On Premise domain can not be empty.'
+            });
+            return;
+        }
+        this.setState({errorMessage: ''});
+        this.setTimeoutAuthDone();
+        if (isWebBrowser) {
+            window.open(this.buildOauthUrl(), '_blank');
+        } else {
+            dynamicRequireElectron().shell.openExternal(this.buildOauthUrl());
+        }
+    }
+
+    updateStateWithEvent(e) {
+        this.setState({[e.target.name]: e.target.value});
+    }
+
+    render() {
+        const {errorMessage, status} = this.state;
+        const submitButton = (value) => {
+            return (
+                <button
+                    onClick={(e) => this.chooseServerType(e.target.value)}
+                    className="btn btn-large btn-primary"
+                    value={value}
+                    style={{margin: '10px'}}
+                >
+                    {SERVER_TYPES[value]}
+                </button>
+            );
+        };
+
+        const loginButton = () => {
+            return (
+                <button
+                    className="btn btn-large btn-primary"
+                    style={{display: 'block', margin: 'auto'}}
+                    onClick={() => this.authenticateUser()}
+                >{'Login'}</button>
+            );
+        };
+
+        const chooseServerType = () => {
+            return (
+                <div className="control-group">
+                <h3 className="block-center-heading">
+                    Choose Your Plotly Use Case
+                </h3>
+                    <div className="controls" style={{padding: '20px'}}>
+                        {submitButton(CLOUD)}
+                        {submitButton(ONPREM)}
+                    </div>
+                    <span>
+                        {'Learn more about our services '}
+                        <a href="https://plot.ly/products/cloud/">
+                        {'here'}</a>.
```

Comment:
it's unsemantic to have links to `here` or `click here`. Instead of `Learn more about our services [here]`, you can just do `[Learn more about our services]`. 

References:
- [Smashing Mag](https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/)
- [W3.org](https://www.w3.org/QA/Tips/noClickHere)
- [Coding Horror](https://blog.codinghorror.com/dont-click-here-the-art-of-hyperlinking/)